msal/src/MSAL.Common.props

```diff
@@ -47,16 +47,22 @@
 
   <PropertyGroup Condition="'$(TargetFramework)' == 'netcoreapp1.0' ">
     <DefineConstants>$(DefineConstants);NET_CORE</DefineConstants>
+    <!-- Feature Flags -->
+    <DefineConstants>$(DefineConstants);MSAL_FEATURE_USERNAMEPASSWORD</DefineConstants>
```

Comment:
Technically this is pretty cool, and a lot of libraries do this. But if our goal is to have a unified API across all tfms, we'll have to include each and every feature to each tfm, and hide some APIs from Intellisense / Build.